hs/src/Reach/JSUtil.hs

```diff
@@ -320,6 +444,90 @@ instance HasJSAnnot JSStatement where
+    JSClass _ x y _z r _s t -> JSClass a' (rjsa a' x) y a' r a' (rjsa a' t)
+    JSConstant _ x y -> JSConstant a' (toJSCL (map (rjsa a') $ jscl_flatten x)) y
+    JSContinue _ x y -> JSContinue a' (rjsa a' x) (rjsa a' y)
+    JSDoWhile _ x _ _ r _ t -> JSDoWhile a' (rjsa a' x) a' a' (rjsa a' r) a' (rjsa a' t)
+    JSFor _ _ y _ r _ t _ v -> JSFor a' a' (rjsaJSL a' y) a' (rjsaJSL a' r) a' (rjsaJSL a' t) a' (rjsa a' v)
+    JSForIn _ _ y z r _s t -> JSForIn a' a' (rjsa a' y) (rjsa a' z) (rjsa a' r) a' (rjsa a' t)
+    JSForVar _ _ _ z _ s _ u _ w -> JSForVar a' a' a' (rjsaJSL a' z) a' (rjsaJSL a' s) a' (rjsaJSL a' u) a' (rjsa a' w)
+    JSForVarIn _ _ _ z r s _ u -> JSForVarIn a' a' a' (rjsa a' z) r (rjsa a' s) a' (rjsa a' u)
+    JSForLet _ _ _ z _ s _ u _ w -> JSForLet a' a' a' (rjsaJSL a' z) a' (rjsaJSL a' s) a' (rjsaJSL a' u) a' (rjsa a' w)
+    JSForLetIn _ _x _ z r s _ u -> JSForLetIn a' a' a' (rjsa a' z) (rjsa a' r) (rjsa a' s) a' (rjsa a' u)
+    JSForLetOf _ _x _ z r s _ u -> JSForLetOf a' a' a' (rjsa a' z) (rjsa a' r) (rjsa a' s) a' (rjsa a' u)
+    JSForConst _ _x _y z _r s _t u _v w -> JSForConst a' a' a' (rjsaJSL a' z) a' (rjsaJSL a' s) a' (rjsaJSL a' u) a' (rjsa a' w)
+    JSForConstIn _ _ _ z r s _ u -> JSForConstIn a' a' a' (rjsa a' z) r (rjsa a' s) a' (rjsa a' u)
+    JSForConstOf _ _ _ z r s _ u -> JSForConstOf a' a' a' (rjsa a' z) r (rjsa a' s) a' (rjsa a' u)
+    JSForOf _ _ y z r _ t -> JSForOf a' a' (rjsa a' y) (rjsa a' z) (rjsa a' r) a' (rjsa a' t)
+    JSForVarOf _ _ _ z r s _ u -> JSForVarOf a' a' a' (rjsa a' z) (rjsa a' r) (rjsa a' s) a' (rjsa a' u)
+    JSAsyncFunction _ _x y _ r _ t u -> JSAsyncFunction a' a' y a' (rjsaJSL a' r) a' (rjsa a' t) (rjsa a' u)
+    JSFunction _ x _ z _ s t -> JSFunction a' (rjsa a' x) a' (rjsaJSL a' z) a' (rjsa a' s) (rjsa a' t)
+    JSGenerator _ _ y _ r _ t u -> JSGenerator a' a' (rjsa a' y) a' (rjsaJSL a' r) a' (rjsa a' t) (rjsa a' u)
+    JSIf _ _x y _ r -> JSIf a' a' (rjsa a' y) a' (rjsa a' r)
+    JSIfElse _ _ y _ r _ t -> JSIfElse a' a' (rjsa a' y) a' (rjsa a' r) a' (rjsa a' t)
+    JSLabelled x _ y -> JSLabelled (rjsa a' x) a' (rjsa a' y)
+    JSEmptyStatement _ -> JSEmptyStatement a'
+    JSExpressionStatement e x -> JSExpressionStatement (rjsa a' e) (rjsa a' x)
+    JSAssignStatement e x y z -> JSAssignStatement (rjsa a' e) (rjsa a' x) (rjsa a' y) (rjsa a' z)
+    JSMethodCall r _ x _ z -> JSMethodCall (rjsa a' r) a' (rjsaJSL a' x) a' (rjsa a' z)
+    JSReturn _ x y -> JSReturn a' (fmap (rjsa a') x) (rjsa a' y)
+    JSSwitch _ _ y _ _ s _ u -> JSSwitch a' a' (rjsa a' y) a' a' s a' (rjsa a' u)
+    JSThrow _ x y -> JSThrow a' (rjsa a' x) (rjsa a' y)
+    JSTry _ x y z -> JSTry a' (rjsa a' x) (map (rjsa a') y) (rjsa a' z)
+    JSVariable _ x y -> JSVariable a' (rjsaJSL a' x) (rjsa a' y)
+    JSWhile _ _ y _ r -> JSWhile a' a' (rjsa a' y) a' (rjsa a' r)
+    JSWith _ _ y _ r s -> JSWith a' a' (rjsa a' y) a' (rjsa a' r) (rjsa a' s)
```

Comment:
I think you should do something like `where r = rjsa a'` and then you'll be able to write stuff like `JSFunction a' (r x) a' (r z) a' (r s) (r t)`